src/components/radio/RadioSelect.tsx

```diff
@@ -1,19 +1,49 @@
 import * as React from 'react';
+import * as _ from 'underscore';
 import {ToggleForm} from '../childForm/ToggleForm';
 import {Radio} from './Radio';
 
 export interface IRadioSelectProps {
+    id?: string;
     name?: string;
-    value?: string;
     disabled?: boolean;
-    onChange?: (value: string) => void;
     children?: Array<React.ReactElement<Radio>> | Array<React.ReactElement<ToggleForm>>;
+    /**
+     * used in RadioSelectConnected component only
+     */
+    valueOnMount?: string;
+    disabledValuesOnMount?: string[];
+}
+
+export interface IRadioSelectDispatchProps {
+    onMount?: (id: string, valueOnMount: string, disabledValues: string[]) => void;
+    onUnmount?: (id: string) => void;
+    onChange?: (value: string, id?: string, e?: React.MouseEvent<HTMLElement>) => void;
+}
+
+export interface IRadioSelectStateProps {
+    value?: string;
+    disabledValues?: string[];
 }
 
-export class RadioSelect extends React.Component<IRadioSelectProps, any> {
-    private handleToggle(value: string) {
+export interface IRadioSelectAllProps extends IRadioSelectProps, IRadioSelectDispatchProps, IRadioSelectStateProps {}
+
+export class RadioSelect extends React.Component<IRadioSelectAllProps, any> {
+    private handleToggle(value: string, e: React.MouseEvent<HTMLElement>) {
         if (this.props.onChange) {
-            this.props.onChange(value);
+            this.props.onChange(value, this.props.id, e);
+        }

```

Comment:
Very facultative suggestion for the 3 `if` in this file:
`callIfDefined(this.props.onChange, value, this.props.id, e);`